### PR TITLE
Test Everything (and small printing change)

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: src/nucleus/tt.ml [hunk 1]
```diff
@@ -641,8 +641,9 @@ and print_share ~penv lshare ppf = match lshare with
 and print_shares ~penv lshares ppf = match lshares with
   | [] -> ()
   | [lshare] -> print_share ~penv lshare ppf
-  | ((_,Inl x) as lshare) :: lshares ->
-    Format.fprintf ppf "%t and %t" (print_share ~penv lshare) (print_shares ~penv:(add_forbidden x penv) lshares)
+  | (l,Inl x) :: lshares ->
+    let x = Name.refresh penv.forbidden x in
+    Format.fprintf ppf "%t and %t" (print_share ~penv (l,Inl x)) (print_shares ~penv:(add_forbidden x penv) lshares)
```
Comment (andrejbauer) [inline src/nucleus/tt.ml]: In general, whenever you print a space you should give the formatter a chance to break the line by printing `@ ` (`@` followed by a space), so `"%t@ and@ %t"` here, and also in `print_share` there is an opportunity to put in `@ ` around `=`.

File: src/nucleus/tt.ml [hunk 2]
```diff
@@ -661,7 +662,7 @@ and print_structure ?max_level ~penv (s,shares) es ppf =
   let rec fold acc es ls shares = match ls, shares with
     | [], [] -> List.rev acc
     | _::ls, (Inr _) :: shares -> fold acc es ls shares
-    | l::ls, (Inl _)::shares ->
+    | l::ls, (Inl _) :: shares ->
```
Comment (andrejbauer) [inline src/nucleus/tt.ml]: This isn't the right place to comment this, but can you please rename the `sum` type and the `Inr` and `Inl` to something more descriptive? It's only used for sharing, so the `Inl` and `Inr` should be called something else that described their role.

Comment (SkySkimmer) [inline src/nucleus/tt.ml]: Some thing like
```
type ('a,'b) constraint_desc =
  | Unconstrained of 'a
  | Constrained of 'b
```
?

Comment (andrejbauer) [inline src/nucleus/tt.ml]: Sounds good to me.

**Pull request merged**